js/source/geojson_source.js

```diff
@@ -41,28 +44,13 @@ module.exports = GeoJSONSource;
  * map.removeSource('some id');  // remove
  */
 function GeoJSONSource(options) {
-    options = options || {};
+    this.options = options = options || {};
```

Comment:
I'd prefer to not save the `options` object. Doing so

 - doesn't make much sense semantically 
 - creates duplicate data within the class
 - opens up the possibility of the `options`  object being modified from the outside 

Instead, you can write `this.type = options.type`